plugin-server/src/main/ingestion-queues/kafka-queue.ts

```diff
@@ -20,18 +20,27 @@ export class KafkaQueue {
     public pluginsServer: Hub
     public workerMethods: WorkerMethods
     public consumerReady: boolean
+    private lastHeartbeat: number | undefined
     private kafka: Kafka
     private consumer: Consumer
+    private consumerConfig: Partial<ConsumerConfig>
     private consumerGroupMemberId: string | null
     private wasConsumerRan: boolean
     private ingestionTopic: string
     private eventsTopic: string
     private eachBatch: Record<string, EachBatchFunction>
 
-    constructor(pluginsServer: Hub, workerMethods: WorkerMethods) {
+    constructor(pluginsServer: Hub, workerMethods: WorkerMethods, consumerConfig: { sessionTimeout: 30000 }) {
         this.pluginsServer = pluginsServer
         this.kafka = pluginsServer.kafka!
-        this.consumer = KafkaQueue.buildConsumer(this.kafka, this.consumerGroupId())
+        this.consumerConfig = consumerConfig
+        this.consumer = KafkaQueue.buildConsumer(this.kafka, this.consumerGroupId(), consumerConfig)
+
+        // Record when the last heartbeat was, so we can check how alive the
+        // `KafkaQueue` is
+        this.lastHeartbeat = undefined
+        this.consumer.on(this.consumer.events.HEARTBEAT, ({ timestamp }) => (this.lastHeartbeat = timestamp))
```

Comment:
Doing side-effects in constructor is in bad taste. Please move to within `start()`